src/components/modules/InfoBox/index.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { PureComponent } from 'react';
+import { StyleSheet, View, Text, ImageBackground, Dimensions } from 'react-native';
+import { ModuleBox } from '@/components/modules';
+import { Image, Container, HtmlText } from '@/components/primitives';
+import { Responsive } from '@/components/utils';
+import { COLOR_MAP } from '@/constants';
+import { BREAKPOINTS } from '@/constants/index';
+import { InfoBoxData, OFFSET_DIRECTION } from '@/types';
+
+interface Props {
+    data: InfoBoxData[];
+}
+
+type State = {
+    readText: string;
+    width: number;
+    height: number;
+    windowWidth: number;
+    imgHeight: number;
+}
+
+export default class InfoBox extends PureComponent<Props, State> {
+    state = {
+        readText: 'Read More',
```

Comment:
instead of storing the readText in state, it'd be cleaner to store the state as a boolean called isExpanded or something like that. Then you can derive the readText where you need it.

Reply:
Changed to isExpanded: boolean